Adds the next 3 upcoming events and webinars to the index page

b-m-f opened the pull request:
# Done
- Adds the next 3 upcoming events and webinars to the index page

Fixes https://github.com/canonical-web-and-design/base-squad/issues/520

## QA
- Use https://github.com/canonical-web-and-design/www.ubuntu.com and replace `canonicalwebteam.blog==1.5.X` with `git+https://github.com/b-m-f/blog-extension@add-upcoming-to-index#egg=canonicalwebteam.blog` in `requirements.txt`
- `./run` the site
- add `{{upcoming}}` to `blog/index.html` and make sure that the data is present when going to `/blog`. It should be 3 entries

jpmartinspt commented:
What about the flask view? It seems to be drifting apart from the django view.
Also it's probably worth having a way of activating the upcoming feature so the request is not made when it's not needed?

b-m-f commented:
For the Flask views I added https://github.com/canonical-web-and-design/canonicalwebteam.blog/issues/61.

I have added a parameter to disable the upcoming posts though.

jpmartinspt commented on canonicalwebteam/blog/django/views.py:
```diff
@@ -47,6 +47,19 @@ def index(request):
+            if upcoming:
+                events = api.get_category_by_slug("events")
+                webinars = api.get_category_by_slug("webinars")
+                upcoming, _ = api.get_articles(
```
We should name this differently from the function argument, or rename the argument.

jpmartinspt commented on canonicalwebteam/blog/django/views.py:
```diff
@@ -47,6 +47,19 @@ def index(request):
+                    per_page=3,
+                    categories=[events["id"], webinars["id"]],
+                )
+            else:
```
Could be declared up top as an empty list to avoid the else.
